src/MvvmNano.Forms/MvvmNanoFormsPresenter.cs

```diff
@@ -90,21 +92,39 @@ public void SetApplication(Application application)
                 .ToArray();
         }
 
+        bool IsMasterDetailApplication()
+        {
+            return Application is MvvmNanoMasterDetailApplication;
+        }
+
+        public void ChangeRootViewModel<TViewModel>() where TViewModel : MvvmNanoViewModel
+        {
+            var viewName = GetViewNameByViewModel(typeof(TViewModel));
+            Type viewType = GetViewTypeByName(viewName); 
+            if(IsMasterDetailApplication() && viewType.GetTypeInfo().IsSubclassOf(typeof(NanoMasterDetailPage)))
+                ((MvvmNanoMasterDetailApplication)Application).SetUpMasterDetailPage<TViewModel>();    
```

Comment:
I agree with you, i will make the `SetUpMainPage` virtual and override it in `MvvmNanoMasterDetailApplication`.